### DISCUSSION TITLE
Prefilling the v2 picker with `*` includes word boundary metacharacter, and matches nothing

### DISCUSSION THREAD

**Discussion by href** [helix-editor/helix#12279] [Troubleshooting] [answered]
### Summary

With the old picker, I often used `*` to use the current selection as search term. This no longer works with the new picker, because it seems to add a word boundary that never matches.

### Reproduction Steps

In the helix source repository:

1. `hx helix-core/src/auto_pairs.rs`
2. Select `pub`.
3. Enter `*`.
4. Open the global search.
5. Even though `pub` is everywhere in the source, nothing is found.

If I use `CTRL+R`, followed by `.` instead, the word is pasted correctly and the search works.

### Platform

macOS

### Terminal Emulator

alacritty

### Installation Method

brew --head

### Helix Version

helix 24.7 (99fdbce5)

**Comment by David-Else** [comment, marked as answer]
It works if you press `enter` afterwards to confirm in the picker.

![image](https://github.com/user-attachments/assets/2f92e2b9-d3f1-45ce-8905-55759b848e92)

**Reply by the-mikedavis** [reply]
Yep you need to "accept" the prefill. It's basically the same thing as when you subselect with `s` for example - the prompt prefills with the contents of the `/` register and you can accept it with Enter

**Reply by the-mikedavis** [reply]
The word boundary part is unrelated, see https://github.com/helix-editor/helix/pull/12126

**Reply by href** [reply]
I see thank you!

**Reply by the-mikedavis** [reply]
Also worth mentioning: the prefill text is styled with the `"ui.text.inactive"` theme key, so if your theme doesn't style that key it is probably indistinguishable from regular picker/prompt text. I would recommend setting that closer to comment color so it's clearer, for example I have:

![inactive](https://github.com/user-attachments/assets/d385243f-c932-4488-92ee-5b54a767a283)

**Reply by href** [reply]
That's good to know, bogster does not set that yet from what I can tell.

**Reply by href** [reply]
Oh and the path is differently styled as well in you screenshot, does that need to be set too? I think an underscore is vital for that one (in bogster, it looks the same like the search results).

**Reply by David-Else** [reply]
You need to add these keys for your theme, here is what I added in `Dark_Plus`:
```
"ui.picker.header.column"    = { underline.style = "line" }
"ui.picker.header.column.active" = { fg ="white", underline.style = "line" }
```
